qcodes/utils/dataset/doNd.py

```diff
@@ -483,6 +484,242 @@ def do2d(
     return _handle_plotting(dataset, do_plot, interrupted())
 
 
+class _AbstractSweep:
```

Comment:
I think we should make this an abstract base class https://docs.python.org/3/library/abc.html

Reply:
Thanks. I will look into using it.